### PR TITLE
Shorten `generation_time_opts()` to `gt_opts()` as alias

### LINKED ISSUE #564: create `gt_opts` as an alias for `generation_time_opts`
As we move towards extending the `_opts` interface to specifying the data and preprocessing (#346) we are adding complexity to the user interaction. Where in the past a user would have written something like:

```r
epinow(reported_cases = cases, generation_time = list(mean = 3, sd = 1))
```
we're moving towards requiring
```r
epinow(
  data = data_opts(cases),
  generation_time = generation_time_opts(Gamma(mean = 3, sd = 1)))
)
```

I'm wondering if we should provide a simpler version where if any of the arguments requiring an `_opts` helper function are not specified using said helper functions we should call it for the user and pass on the first argument. This would enable the user to specify some of their desired (e.g. case data, generation time) without being exposed to the full interface.

This would mean that one could write
```r
epinow(data = cases, generation_time = Gamma(mean = 3, sd = 1))
```
with the same results as the calls above.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Description

This PR closes #564.

The `generation_time_opts()` function was deemed too long and tedious to specify so has been shortened to `gt_opts()`. Calls to both functions are equivalent and they share the same documentation.

## Initial submission checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [ ] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [x] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [x] I have added a news item linked to this PR.

## After the initial Pull Request 

- [ ] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 - `epinow()` now returns the "timing" output in a "time difference"" format that is easier to understand and work with. By @jamesmbaazam in #688 and reviewed by @sbfnk.
 
+## Documentation
+
+- `generation_time_opts()` has been shortened to `gt_opts()` to make it easier to specify. Calls to both functions are equivalent. By @jamesmbaazam in # and reviewed by @<REVIEWER>.
```
Comment (jamesmbaazam) [inline NEWS.md:9]:
```suggestion
- `generation_time_opts()` has been shortened to `gt_opts()` to make it easier to specify. Calls to both functions are equivalent. By @jamesmbaazam in #698 and reviewed by @<REVIEWER>.
```

**Comment by jamesmbaazam** [comment]
CI errors will be fixed when #700 is merged.

**Comment by jamesmbaazam** [comment]
Leaving a note here that `touchstone` has to checkout `main` to perform the benchmarks against this branch, but `main` doesn't have `gt_opts()` yet, so the touchstone script (https://github.com/epiforecasts/EpiNow2/blob/main/touchstone/script.R) here will have to use the long form `generation_time_opts()` for now and be updated when this PR is merged.

**Comment by jamesmbaazam** [comment]
> CI errors will be fixed when #700 is merged.

This has been resolved but was one of two errors in the CI.

**Review by seabbs — APPROVED** [review]
Lgtm

File: NEWS.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 - `epinow()` now returns the "timing" output in a "time difference"" format that is easier to understand and work with. By @jamesmbaazam in #688 and reviewed by @sbfnk.
 
+## Documentation
+
+- `generation_time_opts()` has been shortened to `gt_opts()` to make it easier to specify. Calls to both functions are equivalent. By @jamesmbaazam in #698 and reviewed by @<REVIEWER>.
```
Comment (sbfnk) [inline NEWS.md:9]:
```suggestion
- `generation_time_opts()` has been shortened to `gt_opts()` to make it easier to specify. Calls to both functions are equivalent. By @jamesmbaazam in #698 and reviewed by @seabbs and @sbfnk .
```

File: R/estimate_infections.R [hunk 3]
```diff
@@ -22,7 +22,8 @@
 #'
 #' @param reported_cases Deprecated; use `data` instead.
 #'
-#' @param generation_time A call to [generation_time_opts()] defining the
+#' @param generation_time A call to [gt_opts()] or [generation_time_opts()]
```
Comment (sbfnk) [inline R/estimate_infections.R:25]:
```suggestion
#' @param generation_time A call to [gt_opts()] (or its alias [generation_time_opts()])
```
trying to make clear here that they're the same

**Review by sbfnk — APPROVED** [review]

**Review by sbfnk — DISMISSED** [review]

**Review by sbfnk — DISMISSED** [review]

**Review by sbfnk — APPROVED** [review]

**Pull request merged**